flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/types/ResourceProfile.java

```diff
@@ -367,13 +348,22 @@ public int hashCode() {
 		return result;
 	}
 
+	/**
+	 * Note: Avoid to use this method to do resource checking,
+	 * unless a strict comparison is desired regarding the double values.
+	 * Use {@link #isMatching(ResourceProfile)} or {@link #hasSameResources(ResourceProfile)}
+	 * instead to compare resources ignoring smaller deltas of double values.
```

Comment:
this seems inaccurate; we aren't ignoring deltas; the different between equals and isMatching boils down to specifics of BigDecimal comparisons, which are unrelated to deltas and more about value vs representation.

What's our use-case for making this distinction? Seems like we actually always want to use `hasSameResources`. This seems like we're leaking implementation details to the API.

Reply:
This is a really good question. 

This doc is outdated since the initial version introduces a `ResourceValue` class to manage the double value and its precision. `#equals` needs to retain the transitive property but resource checking ignoring deltas can not ensure that (possibly abs(a -b) < delta, abs(b - c) < delta, but abs(a - c) > delta) . That's why we have to introduce a separate `#hasSameResources`.

However, with BigDecimal, we can achieve that since we can do strict value comparisons. I will drop the change for #equal usages and change #equal implementation instead.

Reply:
done by changing Resource#equals to compare BigDecimal values ignoring the scales. The commit to replace usages of ResourceProfile#equal is also dropped.